lib/getReleases.js

```diff
@@ -15,14 +15,51 @@ class GitHubAPIError extends Error {
   }
 }
 
-exports.findLatestVSIXRelease = async function (/** @type {string} */ repository, /** @type {string} */ version) {
+exports.findLatestVSIXRelease = async function (/** @type {string} */ repository, /** @type {string} */ version, /** @type {string | undefined} */ msVersion,) {
   const [owner, repo] = new URL(repository).pathname.slice(1).split("/");
   console.log(`Fetching ${owner}/${repo} to check for updates...`);
   const token = process.env.GITHUB_TOKEN;
   if (!token) {
     throw new Error("GitHub personal access token is missing");
   }
 
+  if (msVersion) {
+    const release = await fetch("https://api.github.com/graphql", {
+      method: 'POST',
+      headers: { 'Content-Type': 'application/json', 'Authorization': `Bearer ${token}` },
+      body: JSON.stringify({
+        query: `query GetVSIXFromRelease {
+          repository(name: "${repo}", owner: "${owner}") {
+            release(tagName: "${msVersion}") {
+              id
+              tagName
+              releaseAssets(first: 20) {
+                edges {
+                  node {
+                    downloadUrl
+                  }
+                }
+              }
+            }
+          }
+        }` }),
+    });
+    if (!release.ok) {
+      const errorMessage = await release.json();
+      throw new GitHubAPIError(`${errorMessage.message} (HTTP ${release.status})`);
+    }
+
+    const releaseData = await release.json();
+
+    if (releaseData?.data.repository?.release) {
+    return releaseData?.data.repository?.release.releaseAssets.edges.
+      map((/** @type {{ node: {downloadUrl: any;} }} */ a) => a.node.downloadUrl).
+      find((/** @type {string} */ d) => d.match(/\/releases\/download\/[-._a-zA-Z0-9\/%]*\.vsix/g));
+    } else {
+      console.warn("Specified release not found, defaulting to latest release...");
```

Comment:
if msVersion is specified we should make sure that we don't use anything else, so throw or return undefined here.